_posts/2020-07-31-fnn-vae-for-noisy-timeseries/fnn_vae_for_noisy_timeseries.Rmd

```diff
@@ -0,0 +1,509 @@
+was needed for the LSTM: At noise levels 2 and 2.5, that multiplier was set to 5.
+
+As a result, in all cases, there was one latent variable with high variance and a second one of minor importance. For all
+others, variance was close to 0.
+
+*In all cases* here means: In all cases where FNN regularization was used. As already hinted at in the introduction, the main
+regularizing factor providing robustness to noise here seems to be FNN loss, not KL divergence. So for all noise levels,
+besides FNN-regularized LSTM and VAE models we also tested their non-constrained counterparts.
+
+#### Low noise
+
+Seeing how all models performed bravourously on the original deterministic series, a noise level of 1 can almost be treated as
+a baseline. Here you see sixteen 120-timestep predictions from both regularized models, FNN-VAE (dark blue) and FNN-LSTM
+(orange). The noisy test data, both input (`x`, 120 steps) and output (`y`, 120 steps) are displayed in (blue-ish) grey. In
+green, also spanning the whole sequence, we have the original Roessler data, the way they would look had no noise been added.
+
+```{r, eval=TRUE, echo=FALSE, layout="l-page", fig.cap = "Roessler series with added Gaussian noise of standard deviation 1. Grey: actual (noisy) test data. Green: underlying Roessler system. Orange: Predictions from FNN-LSTM. Dark blue: Predictions from FNN-VAE."}
+knitr::include_graphics("images/noise_1.png")
+```
+
+Despite the noise, forecasts from both models look excellent. Is this due to the FNN regularizer?
+
+Looking at forecasts from their unregularized counterparts, we have to admit these do not look any worse. (For better
+comparability, the sixteen sequences to forecast were initiallly picked at random, but used to test all models and
+conditions.)
+
+```{r, eval=TRUE, echo=FALSE, layout="l-page", fig.cap = "Roessler series with added Gaussian noise of standard deviation 1. Grey: actual (noisy) test data. Green: underlying Roessler system. Orange: Predictions from unregularized LSTM. Dark blue: Predictions from unregularized VAE."}
+knitr::include_graphics("images/noise_1_nofnn.png")
+```
+
+What happens when we start to add noise?
+
+#### Substantial noise
+
+Between noise levels 1.5 and 2, something changed, or became noticeable from visual inspection. Let's directly jump to the
+highest-used level though: 2.5.
+
+Here first are predictions obtained from the unregularized models.
+
+```{r, eval=TRUE, echo=FALSE, layout="l-page", fig.cap = "Roessler series with added Gaussian noise of standard deviation 2.5. Grey: actual (noisy) test data. Green: underlying Roessler system. Orange: Predictions from unregularized LSTM. Dark blue: Predictions from unregularized VAE."}
+knitr::include_graphics("images/noise_2.5_nofnn.png")
+```
+
+Both LSTM and VAE, but the latter to an even higher degree, get "distracted" a bit too much by the noise, leading to cases
```

Comment:
Here's another one where you might move the caveat to _after_ the main clause, e.g., _Both LSTM and VAE get "distracted" a bit too much by the noise, the latter to an even higher degree. This leads to cases where…_

Reply:
interesting - same thing as always, that "focus" thing which probably is due to some exaggerated wish to control attention ..